### PR TITLE
Created Faculty activity

### PR DESCRIPTION
**Pull request by zaiverse**

### CODE REVIEW

File: app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            .check(ViewAssertions.matches(ViewMatchers.withText(R.string.weather_card_title)))
+    }
+
+    @Test
+    fun zipCodeText_and_enterBtn_Test() {
+        Espresso.onView(ViewMatchers.withId(R.id.zipcode_faculty))
+            .perform(ViewActions.typeText("12345"), ViewActions.closeSoftKeyboard())
+        Espresso.onView(ViewMatchers.withId(R.id.zipcode_faculty_enter_button)).perform(ViewActions.click())
+        // Assertions or verifications after button click
+    }
```
Comment (emilyyankan) [inline app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt:31]: This test looks incomplete. Comment stated assertions or versification after button click...

File: app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+    }
+
+    @Test
+    fun logoutButtonClickTest() {
+        Espresso.onView(ViewMatchers.withId(R.id.logoutButton_faculty)).perform(ViewActions.click())
+        // Assertions to verify logout behavior, like returning to the login screen
+        // For example, if logout navigates to a login screen:
+        // onView(withId(R.id.login_screen_id)).check(matches(isDisplayed()))
+    }
```
Comment (emilyyankan) [inline app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt:53]: Same here. Assertions statements are missing. `Espresso.onView(ViewMatchers.withId(R.id.logoutButton_faculty)).perform(ViewActions.click())` stimulates the user clicking the logout button in the app. What happens after it logout? does it switch back to the login page?

File: app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import org.junit.Rule
+import org.junit.Test
+
+class FacultyActivityTest {
+    @get:Rule
+    val activityRule = ActivityScenarioRule(FacultyActivity::class.java)
+
+    @Test
+    fun forecastCardTitle_isDisplayed() {
+        Espresso.onView(ViewMatchers.withId(R.id.forecast_faculty_card_title))
```
Comment (emilyyankan) [inline app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt:20]: `Espresso.onView` You can use qualifying alias here. You do not need to include `Espresso` every time.

File: app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Test
+    fun logoutButtonClickTest() {
+        Espresso.onView(ViewMatchers.withId(R.id.logoutButton_faculty)).perform(ViewActions.click())
+        // Assertions to verify logout behavior, like returning to the login screen
+        // For example, if logout navigates to a login screen:
+        // onView(withId(R.id.login_screen_id)).check(matches(isDisplayed()))
+    }
+}
```
Comment (emilyyankan) [inline app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt:54]: Missing empty line here

File: app/src/main/java/com/example/mainactivity/activities/FacultyActivity.kt [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        super.onCreate(savedInstanceState)
+        binding = ActivityFacultyBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        // Set custom toolbar as the support action bar
+        setSupportActionBar(binding.toolbarFaculty)
+
+        // Add logic for the toolbar logout button here:
+        binding.logoutButtonFaculty.setOnClickListener {
+            // Handle the logout logic here
```
Comment (emilyyankan) [inline app/src/main/java/com/example/mainactivity/activities/FacultyActivity.kt:26]: missing Logout button logic..

**Review by emilyyankan — CHANGES_REQUESTED** [review]
Left a few comments

**Review by emilyyankan — APPROVED** [review]

File: app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt (hunk 1 above)
Comment (zaiverse) [reply app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt:31]: Issue Resolved

File: app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt (hunk 2 above)
Comment (zaiverse) [reply app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt:53]: Issue resolved

File: app/src/main/java/com/example/mainactivity/activities/FacultyActivity.kt (hunk 5 above)
Comment (zaiverse) [reply app/src/main/java/com/example/mainactivity/activities/FacultyActivity.kt:26]: Issue resolved

File: app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt (hunk 4 above)
Comment (zaiverse) [reply app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt:54]: I will make sure to add that with my next PR

File: app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt (hunk 3 above)
Comment (zaiverse) [reply app/src/androidTest/java/com/example/mainactivity/FacultyActivityTest.kt:20]: Made note of  it thank you

**Pull request merged**